stem/solver.py

```diff
@@ -1,10 +1,488 @@
+        - rebuild_level (int): 2 if the lhs matrix is rebuilt at each non-linear iteration, 1 if the lhs matrix \
+            is rebuilt at each time step, 0 if the lhs matrix is only built once
+        - prebuild_dynamics (bool): if true, the mass and damping matrices are prebuilt and directly used to calculate \
+            the rhs. If false, the mass and damping matrices are built at each non-linear iteration for calculating \
+            the rhs
+        - convergence_criteria (:class:`ConvergenceCriteriaABC`): convergence criteria, \
+            :class:`DisplacementConvergenceCriteria`, :class:`ResidualConvergenceCriteria`, \
+            :class:`WaterPressureConvergenceCriteria` or :class:`DisplacementAndWaterPressureConvergenceCriteria`
+        - reset_displacements (bool): if true, the displacements are reset at the beginning of the phase
+        - strategy_type (:class:`StrategyTypeABC`): strategy type, :class:`NewtonRaphsonStrategy`, \
+            :class:`LineSearchStrategy` or :class:`ArcLengthStrategy`. Default value is :class:`NewtonRaphsonStrategy`.
+        - scheme (:class:`SchemeABC`): scheme, :class:`NewmarkSceme` or :class:`BackwardEulerScheme`. Default value \
+            is :class:`NewmarkSceme`.
+        - linear_solver_settings (:class:`LinearSolverSettingsABC`): linear solver settings, currently only \
+            :class:`Amgcl` is supported
+        - rayleigh_m (Optional[float]): mass proportional damping parameter
+        - rayleigh_k (Optional[float]): stiffness proportional damping parameter
+
+    """
+
+    solution_type: SolutionType
+    time_integration: TimeIntegration
+
+    rebuild_level: int
+    prebuild_dynamics: bool
+
+    convergence_criteria: ConvergenceCriteriaABC
+    reset_displacements: bool = False
+    strategy_type: StrategyTypeABC = field(default_factory=NewtonRaphsonStrategy)
+    scheme: SchemeABC = field(default_factory=NewmarkScheme)
+    linear_solver_settings: LinearSolverSettingsABC = field(default_factory=Amgcl)
+    rayleigh_m: Optional[float] = None
+    rayleigh_k: Optional[float] = None
+
+    def __post_init__(self):
+        """
+        Post initialization method
+
+        Raises:
+            ValueError: if the rayleigh damping parameters are not provided for dynamic analysis
+        """
+        if self.solution_type == SolutionType.DYNAMIC:
+            if self.rayleigh_m is None or self.rayleigh_k is None:
+                raise ValueError("Rayleigh damping parameters must be provided for dynamic analysis")
+
+
+@dataclass
+class Problem:
+    """
+    Class containing information about the problem settings and the solver settings
+
+    Attributes:
+        - problem_name (str): name of the problem
+        - number_of_threads (int): number of threads used for the analysis
+        - settings (:class:`SolverSettings`): dictionary containing the solver settings
+        - echo_level (int): echo level. Default value is 1.
```

Comment:
what does echo level mean? what are the options?

Reply:
0 , 1, or 2, the higher the level the more Kratos response is written in the console regarding the solver